R/object_overwrite_linter.R

```diff
@@ -101,14 +100,17 @@ object_overwrite_linter <- function(
 
     xml <- source_expression$xml_parsed_content
 
-    bad_expr <- xml_find_all(xml, xpath)
-    bad_symbol <- xml_text(xml_find_first(bad_expr, "SYMBOL"))
-    source_pkg <- pkg_exports$package[match(bad_symbol, pkg_exports$name)]
-    lint_message <-
-      sprintf("'%s' is an exported object from package '%s'. Avoid re-using such symbols.", bad_symbol, source_pkg)
+    assigned_exprs <- xml_find_all(xml, xpath_assignments)
+    assigned_symbols <- get_r_string(assigned_exprs, "SYMBOL")
+    is_bad <- assigned_symbols %in% pkg_exports$name
+    source_pkg <- pkg_exports$package[match(assigned_symbols[is_bad], pkg_exports$name)]
+    lint_message <- sprintf(
+      "'%s' is an exported object from package '%s'. Avoid re-using such symbols.",
```

Comment:
Copying here to put it all side-by-side:

> 'fun' overwrites 'pkg::fun'. Avoid overwriting symbols from other packages.

I don't like 'overwrites' which is misleading. 'masks' might be better. OTOH I don't see all that much wrong with the current message. Alternative:

> '%s' is exported by package '%s'. Avoid re-using function names for local symbols.